File: app/src/main/java/de/devisnik/android/bigmouth/SummaryUpdater.java
/**
 * 
 */
package de.devisnik.android.bigmouth;

import java.util.Arrays;

import android.preference.ListPreference;
import android.preference.Preference;

class SummaryUpdater implements Preference.OnPreferenceChangeListener {

	@Override
	public boolean onPreferenceChange(Preference preference, Object newValue) {
		if (preference instanceof ListPreference) {
			updateListSummary((ListPreference)preference, (String) newValue);
		}
		else 
			updateSummary(preference, (String) newValue);
		return true;
	}
	
	void updateListSummary(ListPreference listPreference, String newValue) {
		int entryIndex = Arrays.asList(listPreference.getEntryValues()).indexOf(newValue);
		listPreference.setSummary(listPreference.getEntries()[entryIndex]);	
	}
	
	void updateSummary(Preference listPreference, String newValue) {
		listPreference.setSummary(newValue);	
	}

}

File: app/src/main/kotlin/de/devisnik/android/bigmouth/SummaryUpdater.kt
/**

 */
package de.devisnik.android.bigmouth

import android.preference.ListPreference
import android.preference.Preference

internal class SummaryUpdater : Preference.OnPreferenceChangeListener {

    override fun onPreferenceChange(preference: Preference, newValue: Any): Boolean {
        when (preference) {
            is ListPreference -> updateListSummary(preference, newValue as String)
            else -> updateSummary(preference, newValue as String)
        }
        return true
    }

    fun updateListSummary(listPreference: ListPreference, newValue: String) {
        val entryIndex = listPreference.entryValues.indexOf(newValue)
        listPreference.summary = listPreference.entries[entryIndex]
    }

    fun updateSummary(listPreference: Preference, newValue: String) {
        listPreference.summary = newValue
    }

}
